tests/containers/podman_docker_interop.pm

```diff
@@ -0,0 +1,126 @@
+# SUSE's openQA tests
+#
+# Copyright 2022-2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Validate docker and podman network interoperability
+# - validate if podman and docker networking works together
+# - check if firewalld doesn't break either
+# Maintainer: QE-C team <qa-c@suse.de>
+
+#use Mojo::Base qw(consoletest);
+use Mojo::Base qw(containers::basetest);
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+use version_utils qw(check_os_release get_os_release is_sle is_sle_micro is_transactional);
+use transactional qw(check_reboot_changes);
+#use containers::common;
+
+my $test_img = "registry.opensuse.org/opensuse/nginx";
+my $test_cmd = "curl https://opensuse.org/ -vso/dev/null";
+
+sub run {
+    my ($self, $args) = @_;
+
+    select_serial_terminal;
+
+    # Initialize and cleanup podman
+    my $podman = $self->containers_factory('podman');
+    $podman->cleanup_system_host();
+
+    # Required, otherwise any subsequent outgoing podman connections fail
+    assert_script_run("podman run --rm $test_img $test_cmd");
+
+    # Install firewalld in case it's not installed
+    if (script_run('which firewall-cmd') != 0) {
+        if (is_transactional) {
+            trup_call('pkg install firewalld');
+            check_reboot_changes;
+        } else {
+            zypper_call('in firewalld');
+        }
+    }
+
+    # Initialize and cleanup docker
+    my $docker = $self->containers_factory('docker');
+    $docker->cleanup_system_host();
+
+    # Stop docker daemon and start firewalld
+    systemctl("disable --now docker");
+    systemctl("enable --now firewalld");
+
+    systemctl("is-active docker", expect_false => 1);
+    systemctl("is-active firewalld");
+
+    record_info('Firewalld Zones', script_output("firewall-cmd --zone=public --list-all && firewall-cmd --get-zones | grep docker && firewall-cmd --zone=docker --list-all"));
+
+    script_retry("podman pull $test_img", retry => 3, delay => 60, timeout => 180);
+
+    # Start nginx in podman and validate initial ingress connectivity
+    assert_script_run("podman run -d --name nginx --rm $test_img");
+    my $nginx_podman_ip = script_output("podman container inspect nginx -f '{{ .NetworkSettings.IPAddress }}'");
+    assert_script_run("curl $nginx_podman_ip -svo/dev/null");
+
+    # Validate initial podman egress connectivity
+    assert_script_run("podman run --rm $test_img $test_cmd");
+
+    # Start docker and validate podman connectivity
+    systemctl("start docker");
+    systemctl("is-active docker");
+    systemctl("is-active firewalld");
+    if (script_run("curl $nginx_podman_ip -svo/dev/null") != 0) { bsc_1213811(); }
```

Comment:
Perhaps it's a good idea to record in the `bsc_1213811` routine if ingress or egress fails. We could add an optional argument to that subroutine, that adds the corresponding message to the record_soft_failure.

e.g. something like

```suggestion
    if (script_run("curl $nginx_podman_ip -svo/dev/null") != 0) { bsc_1213811("egress failed"); }
```